src/context/myContext.tsx

```diff
@@ -0,0 +1,118 @@
+import
+React, {
+  createContext, useContext, useEffect, useState, ReactNode,
+}
+  from 'react';
+import { getTodos, getUser } from '../api';
+import { User } from '../types/User';
+
+interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user?: User;
+}
+
+interface MyContextProps {
+  todos: Todo[];
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  TodosFromServer: Todo[];
+  isLoad: boolean;
+  handleCompletedTodos: (sortType: string) => void;
+  searchTodo: (value: string) => void;
+  activeTodo: Todo | null;
+  setActiveTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+}
+
+const MyContext = createContext<MyContextProps | undefined>(undefined);
+
+interface MyContextProviderProps {
+  children: ReactNode;
+}
+
+const MyContextProvider: React.FC<MyContextProviderProps> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [TodosFromServer, setAllTodos] = useState<Todo[]>([]);
+  const [isLoad, setIsLoad] = useState(true);
+  const [activeTodo, setActiveTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos().then((allTodos: Todo[]) => {
+      const todoPromises = allTodos.map(todo => {
+        return getUser(todo.userId)
+          .then((user: User) => {
+            const updatedTodo = { ...todo, user };
+
+            return updatedTodo;
+          });
+      });
```

Comment:
you are already using `Promise.all` try to refactor youre code, using `async` and `await` to avoid nesting a loop of promises inside a `then` block
Also, you need to check if the promise throws, add a `catch` block.